softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java

```diff
@@ -6,15 +6,12 @@
 import com.softgroup.common.protocol.Response;
 import com.softgroup.common.protocol.ResponseData;
 
-public class AbstractRequestHandler<T extends RequestData, R extends ResponseData> implements RequestHandler {
-	@Override
-	public String getName() {
-		return null;
-	}
+public abstract class AbstractRequestHandler<R extends RequestData, S extends ResponseData> implements RequestHandler {
 
 	@Override
-	public Response<R> handle(Request<?> msg) {
-		return null;
+	public Response<S> handle(final Request<?> msg) {
+		return commandHandle((Request<R>) msg);
```

Comment:
@RunWith(MockitoJUnitRunner.class)
public class AuthorizationCommandsIT {

    @Test
    public void testCommandLoginHandler() {
        Request<LoginRequestData> request = mock(Request.class);

        Response<LoginResponseData> response = new AuthCmdLoginHandler().handle( request );

        assertThat( response.getStatus().getCode(), is(200) );
        assertThat( response.getStatus().getMessage(), is("OK") );
        assertThat( response.getData().getToken(), is("TOKEN_1") );
    }
}